core/src/main/scala/org/bitcoins/core/crypto/SchnorrNonce.scala

```diff
@@ -0,0 +1,46 @@
+package org.bitcoins.core.crypto
+
+import org.bitcoins.core.protocol.NetworkElement
+import org.bitcoins.core.util.CryptoUtil
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+
+case class SchnorrNonce(bytes: ByteVector) extends NetworkElement {
+  require(bytes.length == 32, s"Schnorr Nonce must be 32 bytes, got $bytes")
+
+  def publicKey: ECPublicKey = Schnorr.computeR(this)
+}
+
+object SchnorrNonce {
+
+  @tailrec
+  def fromBytes(bytes: ByteVector): SchnorrNonce = {
+    require(bytes.length <= 33,
+            s"Schnorr Nonce must be less than 33 bytes, got $bytes")
+
+    if (bytes.length == 32)
+      SchnorrNonce(bytes)
+    else if (bytes.length < 32) {
+      // means we need to pad the private key with 0 bytes so we have 32 bytes
+      SchnorrNonce.fromBytes(bytes.padLeft(32))
+    } else {
```

Comment:
I think it is probably better to be super explicit here and see 

```scala 
else if (bytes.length == 33) {
  //make recursive call
}
```

this protects ourselves from doing something stupid like removing the require in the future. In the else case we should still throw, even if it is redundant.